src/Tmds.Ssh/ECDHKeyExchange.cs

```diff
@@ -99,86 +78,31 @@ private byte[] CalculateExchangeHash(SequencePool sequencePool, SshConnectionInf
         writer.WriteString(public_host_key);
         writer.WriteString(q_c);
         writer.WriteString(q_s);
-        writer.WriteMPInt(sharedSecret);
+        writer.WriteString(sharedSecret);
 
-        using IncrementalHash hash = IncrementalHash.CreateHash(_hashAlgorithmName);
+        using IncrementalHash hash = IncrementalHash.CreateHash(hashAlgorithmName);
         foreach (var segment in sequence.AsReadOnlySequence())
         {
             hash.AppendData(segment.Span);
         }
         return hash.GetHashAndReset();
     }
```

Comment:
Looks like `CalculateExchangeHash` can be shared as well

Reply:
It could with some work that deals with the `ECPoint`. I'm going to merge this PR as it is now. We can tackle this in a separate PR.

Reply:
👍🏼